packages/components/src/Typeahead/Typeahead.component.renderers.js

```diff
@@ -150,6 +142,32 @@ export function renderItemsContainerFactory(
 					shift: {
 						enabled: false,
 					},
+					computePosition: {
+						enabled: true,
+						fn: data => {
+							const GAP = 15; // the offset between the end of items container and screen boundaries
+							const inputDimensions = data.offsets.reference;
+							const { top, height } = inputDimensions;
+							const offsetTop = top - GAP;
+							const offsetBottom = window.innerHeight - top - height - GAP;
+							const placements = data.placement.split('-');
+							let newPlacement = data.placement;
+							if (placements[0] === 'top' && offsetBottom > offsetTop) {
+								newPlacement = `bottom-${placements[1]}`;
+							}
+							const maxHeight = newPlacement.includes('top') ? offsetTop : offsetBottom;
+
+							return {
+								...data,
+								placement: newPlacement,
+								styles: {
+									...data.styles,
+									width: inputDimensions.width,
+									maxHeight,
+								},
+							};
+						},
+					},
```

Comment:
this is an almost pure function (depends on window) so please put it outside as a function

Reply:
done